### ISSUE TITLE
Incorrect session dates for subjects

### ISSUE THREAD

**Issue by botsunny** [connectomicslab/Aneurysm_Detection#6]
Hi there! Thank you for this repository.

I am playing around with the inference code when I realised that the `participants.tsv` file in the dataset downloaded from OpenNeuro contains session dates that do not align with `all_test_subs_inhouse.pkl` in the `extra_files` directory of this repository.

![image](https://github.com/connectomicslab/Aneurysm_Detection/assets/71499037/458be0c7-aba9-4e7c-88ce-0f9b0d623c93)

As seen above, the session date for sub-000 in the OpenNeuro dataset is 20110101 but in the given pkl file, it is 20101230. Same goes for the other pkl files in `extra_files` such as `patients_with_voxelwise_labels.pkl`, `patients_with_weak_labels.pkl` and the pkl files in `Forced_CV_for_reproducibility`.

Therefore, after running the inference step of the pipeline as described in the README, using the available train outputs/weights provided in `extra_files`, inference results were produced for only 6 patients across the 5 folds, as majority of the testing subject-sessions in each fold does not exist in the OpenNeuro dataset at all.

I tried running the pipeline from step 1 and training the network myself, but the  result is the same as `Forced_CV_for_reproducibility` is used to restore the subject-session names.

![image](https://github.com/connectomicslab/Aneurysm_Detection/assets/71499037/e7712d16-fbd4-4653-b35f-d26eae98a38c)

(Notice how the session date for sub-005 in the output of `self_trained` is different from that in `participants.tsv`)

Are the contents of `extra_files` currently outdated with the OpenNeuro dataset? I am not super experienced with this and have not engaged with a project of this scale, so apologies if I am misunderstanding anything!

**Comment by tommydino93** [comment]
Hi @botsunny,
Thank you for your interest here and for pointing out the inconsistency!
You are right, the sessions don't match. The ones from the github repo are outdated. You should use the ones from OpenNeuro (from the dataset itself and from the participants.tsv). However, the subjects should match, so for instance the subjects you find on all_test_subs_inhouse.pkl are correct, though you need to change the sessions.
I currently don't have much time now to correct this. If you manage to create a .pkl file with the correct sessions it would be super useful so I can upload it in the github repo :)
Leaving this open for now as a reminder for the future

**Comment by botsunny** [comment]
Hi there, thank you for the reply!

I had some time to update the session dates within the .pkl files and it seems to be working (with the exception of a couple subjects which are no longer available in the OpenNeuro dataset), though I am running into an (possibly unrelated) issue where `tf.image.per_image_standardization` of the sliding window fails on the patches of some subjects due to "invalid dimension reduction". I will attempt to confirm if the .pkl files are responsible for this before sharing.

Additionally, on an unrelated question, does the model always produce results even if no aneurysms are detected? Would the resulting nii.gz masks simply be empty? Do let me know if I should open another issue for this question.

Thanks again!

**Comment by tommydino93** [comment]
Ok, please let me know if the problem persists after checking the .pkl files.

Regarding your second question, YES the model returns an empty array for patients without aneurysms

**Comment by Ophelia-Wang** [comment]
> Hi there, thank you for the reply!
> 
> I had some time to update the session dates within the .pkl files and it seems to be working (with the exception of a couple subjects which are no longer available in the OpenNeuro dataset), though I am running into an (possibly unrelated) issue where `tf.image.per_image_standardization` of the sliding window fails on the patches of some subjects due to "invalid dimension reduction". I will attempt to confirm if the .pkl files are responsible for this before sharing.
> 
> Additionally, on an unrelated question, does the model always produce results even if no aneurysms are detected? Would the resulting nii.gz masks simply be empty? Do let me know if I should open another issue for this question.
> 
> Thanks again!

Hi @botsunny , thanks a lot for investigating this and for the detailed description.

I’m currently trying to reproduce the paper as well and ran into exactly the same issue: using the pre-trained models and the .pkl files in `extra_files` with the current OpenNeuro release of ds003949-1.0.1, inference only produces results for 6 subjects across the 5 folds, because most of the `sub-ses` combinations in the .pkl files do not exist anymore in the BIDS dataset (the subject IDs match, but the session dates are different, as you described).

You mentioned on Mar 26, 2024 that you had updated the session dates inside the .pkl files and that this seemed to work (except for a few subjects that are no longer available), but that you wanted to confirm things before sharing.

If you are comfortable sharing, would you mind uploading:
- either the updated .pkl files (e.g. `all_test_subs_inhouse.pkl`, `patients_with_weak_labels.pkl`, `patients_with_voxelwise_labels.pkl`, and the ones in `Forced_CV_for_reproducibility`), or
- a small script that you used to remap the session dates based on `participants.tsv`?

This would be extremely helpful for other users who are trying to reproduce the results with the current version of the OpenNeuro dataset. Also, it might be useful to include these updated files (or the script) directly in the repo for reproducibility.

Many thanks again for your work and for any guidance you can provide!
